chia/server/server.py

```diff
@@ -279,7 +279,7 @@ async def start_server(self, on_connect: Callable = None):
         # prefer_ipv6 is set in which case we use the IPv6 port
         #
         if self._port == 0:
-            self._port = select_port(self.root_path, self.runner.addresses)
+            self._port = select_port(self.config.get("prefer_ipv6", False), self.runner.addresses)
```

Comment:
I don't believe this works since prefer_ipv6 is a global config item at the top level of the config, and the config entry passed into the ChiaServer is the specific entry for that service, so this will always be false and never read the actual prefer_ipv6 setting.

Reply:
Sounds right.  `:[`  Thanks for catching this.

Reply:
Thanks! Fixed in 23992f7c3f93f732213a28c44383ec21fe50433e.